### DISCUSSION TITLE
Time switched to Summer time too early

### DISCUSSION THREAD

**Discussion by FastFrank** [jomjol/AI-on-the-edge-device#2166] [General] [answered]
### The Problem

Timezone in Configuration:

CET-1CEST,M3.5.0,M10.5.0/3

Log from MQTT last night: (12-03-2023)

JSON gewijzigd in { "value": "18839.45", "raw": "18839.45", "pre": "18839.45", "error": "no error", "rate": "0.008000", "timestamp": "2023-03-12T03:04:41+0200" }
02:06:01 - 8 uur geleden
JSON gewijzigd in { "value": "18839.41", "raw": "18839.41", "pre": "18839.41", "error": "no error", "rate": "0.004000", "timestamp": "2023-03-12T01:59:41+0100" }
02:01:00 - 8 uur geleden

### Version

15.0.3

### Logfile

```shell
[0d00h00m04s] 2023-03-12T08:52:26	<INF>	[SNTP] TimeServer: nl.pool.ntp.org
[0d00h00m04s] 2023-03-12T08:52:26	<INF>	[SNTP] Configuring NTP Client...
[0d00h00m04s] 2023-03-12T10:52:26	<INF>	[SNTP] Time zone set to CET-1CEST
[0d00h00m04s] 2023-03-12T10:52:26	<INF>	[SNTP] Time is already set: 2023-03-12 10:52:26
```

### Expected Behavior

Change the +1:00 to +02:00 on Month 3 last week day 0

in the log:
[SNTP] Time zone set to CET-1CEST,M3.5.0,M10.5.0/3
instead of:
[SNTP] Time zone set to CET-1CEST

**Comment by caco3** [comment]
did you set the timezone settings correctly?

**Reply by kki03** [reply]
Hi,
I have the same problem. The time changed last night and is now offset by one hour.

According to the docs for Europe / Vienna it should be the following.
```
Region | Code
Europe/Vienna | CET-1CEST,M3.5.0,M10.5.0/3
```
I checked my settings and it is set like that.

**Reply by FastFrank** [reply]
I have the same setting in config.ini but in the log the part after the comma is not reported...

[System]
TimeZone = CET-1CEST,M3.5.0,M10.5.0/3
TimeServer = nl.pool.ntp.org

Log:

[0d00h00m04s] 2023-03-12T11:26:07 <INF> [SNTP] TimeServer: nl.pool.ntp.org
[0d00h00m04s] 2023-03-12T11:26:07 <INF> [SNTP] Configuring NTP Client...
[0d00h00m04s] 2023-03-12T13:26:07 <INF> [SNTP] Time zone set to CET-1CEST
[0d00h00m04s] 2023-03-12T13:26:07 <INF> [SNTP] Time is already set: 2023-03-12 13:26:07

**Reply by caco3** [reply]
hmm, onmy system it is wrong as well.

But it must be a bug in the Espressif framework, as that one manages the NTP config

**Comment by Danit2** [comment]
Same here. 
![image](https://user-images.githubusercontent.com/71522810/224545038-37edcd1d-4b73-4114-9cbd-83374053a7da.jpeg)

**Comment by caco3** [comment, marked as answer]
Please have a, look on https://github.com/jomjol/AI-on-the-edge-device/issues/2168#issuecomment-1465220264

**Comment by joernu76** [comment]
Do I understand correctly, that the "fix" for the one-hour discrepancy is to leave the timezone empty (which incidentally requires a new build to not crash)?

**Reply by caco3** [reply]
no, it was simply a bug in the configuration parser.
Eg, when you had the setting `CET-1CEST,M3.5.0,M10.5.0/3`, it truncated at the `,`, meaning it just used `CET-1CEST`.

BTW, if you keep it empty, it will use the default which is `CET-1CEST,M3.5.0,M10.5.0/3`

**Reply by joernu76** [reply]
Works like a charm after an update to 15.1.

Thanks, great work!
